Project-19-React-router/frontend/src/util/auth.js

```diff
@@ -0,0 +1,40 @@
+import { redirect } from "react-router-dom";
+
+export function getTokenDuration() {
+  const storedExpirationDate = localStorage.getItem("expiration");
+  const expirationDate = new Date(storedExpirationDate);
+  const now = new Date();
+  const duration = expirationDate.getTime() - now.getTime();
+  return duration;
+}
```

Comment:
@sknepali The getTokenDuration function assumes that the stored expiration date can be parsed into a valid date. If it can't, this could lead to unexpected behavior. We should add error handling to account for this.